### ISSUE TITLE
checkbox: aria-checked attribute gives unexpected results

### ISSUE THREAD

**Issue by sharonlevyex** [angular/material#12046]
## Bug Report
In md-checkbox, when unchecking the sub checkboxes the screen reader reads "half checked" on the main checkbox even though it's supposed to read "not checked".  

## Demo and steps to reproduce the issue
**Demo URL***:  go to select all demo in https://material.angularjs.org/1.2.1/demo/checkbox
![image](https://user-images.githubusercontent.com/73985320/98937924-cc66b980-24ef-11eb-8208-085ba47bc4f7.png)

#### Detailed Reproduction Steps
1. uncheck checkbox 1
2. now the select all should be unchecked
3. See that the select all checkbox has aria-checked="mixed" attribute.

### Explain the expected behavior
After unselecting all checkboxes the aria-checked attribute is supposed to be "false".

### Explain the current behavior
After unselecting all checkboxes the aria-checked attribute is "mixed".

**Comment by Splaktar** [comment]
Thank you for this report! I verified this on `1.2.1.` and `HEAD`. the `md-indeterminate` class is removed from the select all checkbox, but the `aria-checked` state is not updated properly.
